semgrep/semgrep/core_runner.py

```diff
@@ -199,99 +199,98 @@ def _run_rule(
+            [rule]
+        ).items():
+            targets = target_manager.get_files(language, rule.includes, rule.excludes)
+            if targets == []:
+                continue
+
+            # semgrep-core doesn't know about OPERATORS.REGEX - this is
+            # strictly a semgrep Python feature. Regex filtering is
+            # performed purely in Python code then compared against
+            # semgrep-core's results for other patterns.
+            patterns_regex, patterns = partition(
+                lambda p: p.expression.operator == OPERATORS.REGEX,
+                all_patterns_for_language,
+            )
+            if patterns_regex:
+                patterns_json = [pattern.to_json() for pattern in patterns_regex]
 
-            for language, all_patterns_for_language in self._group_patterns_by_language(
-                [rule]
-            ).items():
-                targets = target_manager.get_files(
-                    language, rule.includes, rule.excludes
-                )
-                if targets == []:
-                    continue
+                try:
+                    patterns_re = [
+                        (pattern["id"], re.compile(pattern["pattern"]))
+                        for pattern in patterns_json
+                    ]
+                except re.error as err:
+                    raise SemgrepError(f"invalid regular expression specified: {err}")
 
-                # semgrep-core doesn't know about OPERATORS.REGEX - this is
-                # strictly a semgrep Python feature. Regex filtering is
-                # performed purely in Python code then compared against
-                # semgrep-core's results for other patterns.
-                patterns_regex, patterns = partition(
-                    lambda p: p.expression.operator == OPERATORS.REGEX,
-                    all_patterns_for_language,
+                re_fn = functools.partial(get_re_matches, patterns_re)
+                with multiprocessing.Pool(self._jobs) as pool:
+                    matches = pool.map(re_fn, targets)
+
+                outputs.extend(
+                    single_match
+                    for file_matches in matches
+                    for single_match in file_matches
                 )
-                if patterns_regex:
-                    patterns_json = [pattern.to_json() for pattern in patterns_regex]
 
-                    try:
-                        patterns_re = [
-                            (pattern["id"], re.compile(pattern["pattern"]))
-                            for pattern in patterns_json
-                        ]
-                    except re.error as err:
-                        raise SemgrepError(
-                            f"invalid regular expression specified: {err}"
-                        )
+            patterns_json = [p.to_json() for p in patterns]
+            # very important not to sort keys here
```

Comment:
I think we can remove this comment at this point. I'm assuming we're good here since we moved to `ruamel` instead of `yaml`.

Reply:
Good point. Though i think not cause of ruamel change but cause of per-rule running change.